src/unittest/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2022 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+# Author: Luca Tricerri
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+cmake_minimum_required(VERSION 3.5)
+
+include(FetchContent)
+
+#GTEST
+FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY "https://github.com/google/googletest"
+        GIT_TAG        main
+)
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+include(GoogleTest)
+
+project(unittest LANGUAGES CXX)
+
+enable_testing()
+
+add_executable(${PROJECT_NAME})
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+set(CMAKE_INSTALL_RPATH ./)
+
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_20)
+
+target_sources(${PROJECT_NAME}
+    PRIVATE
+    gtest_main.cc
+    testServiceParserForMultipleFT.cpp
+    testDeviceMultipleFTSensors.cpp
+  )
+
+target_link_libraries(${PROJECT_NAME}
+PRIVATE 
+  pthread
```

Comment:
This will not work on Windows, you can use `find_package(Threads)` and then link `Threads::Threads` imported target instead linking directly the `pthread` library, see https://cmake.org/cmake/help/latest/module/FindThreads.html .